src/Tools/Source/RunTests/TestRunner.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace RunTests
+{
+    internal sealed class TestRunner
+    {
+        private struct TestResult
+        {
+            internal readonly bool Succeeded;
+            internal readonly string AssemblyName;
+            internal readonly TimeSpan TimeSpan;
+
+            internal TestResult(bool succeeded, string assemblyName, TimeSpan timeSpan)
+            {
+                Succeeded = succeeded;
+                AssemblyName = assemblyName;
+                TimeSpan = timeSpan;
+            }
+        }
+
+        private readonly string _xunitConsolePath;
+
+        internal TestRunner(string xunitConsolePath)
+        {
+            _xunitConsolePath = xunitConsolePath;
+        }
+
+        internal async Task<bool> RunAll(IEnumerable<string> assemblyList)
+        {
+            var max = Environment.ProcessorCount;
+            var allPassed = true;
+            var waiting = new Stack<string>(assemblyList); 
+            var running = new List<Task<TestResult>>();
+            var completed = new List<TestResult>();
+
+            do
+            {
+                var i = 0;
+                while (i < running.Count)
+                {
+                    var task = running[i];
+                    if (task.IsCompleted)
+                    {
+                        var testResult = await task.ConfigureAwait(false);
```

Comment:
The `await` call will rethrow the inner exception if there is one while Result will throw the aggregate.